app/src/main/java/org/mozilla/focus/utils/AndroidBundleSerializer.java

```diff
@@ -0,0 +1,151 @@
+package org.mozilla.focus.utils;
+
+import android.os.Bundle;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class AndroidBundleSerializer {
+
+    private final Map<String, DataTypeHandler> dataTypeHandlers;
+
+    public interface DataTypeHandler {
+        String getName();
+
+        SerializedItem create(Bundle bundle, String key) throws IOException;
+
+        void restore(Bundle bundle, SerializedItem serializedItem) throws IOException;
+    }
+
+    public AndroidBundleSerializer() {
+        dataTypeHandlers = new HashMap<>();
+
+        register(new ByteArrayHandler());
+        register(new StringHandler());
+        // TODO: support other type of handlers
+    }
+
+    private void register(final DataTypeHandler type) {
+        final String name = type.getName();
+        dataTypeHandlers.put(name, type);
+    }
+
+    public void serializeBundle(ObjectOutputStream outputStream, Bundle bundle) throws IOException {
+        if (bundle != null && bundle.size() > 0) {
+            List<SerializedItem> serializedItemList = new ArrayList<>();
+            Set<String> keys = bundle.keySet();
+            for (String key : keys) {
+                Object value = bundle.get(key);
+                if (value == null) {
+                    continue;
+                }
```

Comment:
In my personal opinion, this is not what I expect if we want to make it general
```java
Bundle before = new Bundle();
before.putString("keyA", "a");
before.putString("keyB", null);
before.contains("keyA"); // return true
before.contains("keyB"); // return true

Bundle after = deserialize(serialize(before));
after.contains("keyA"); // return true
after.contains("keyB"); // return false
```

Reply:
For this case, it seems a NULL handler is needed to serialize/de-serialize the null value.